yarn_api_client/base.py

```diff
@@ -6,43 +6,62 @@
 
 from .errors import APIError, ConfigurationError
 
+try:
+    from urlparse import urlparse, urlunparse
+except ImportError:
+    from urllib.parse import urlparse, urlunparse
 
 class Response(object):
     def __init__(self, response):
         self.data = response.json()
 
-
 class BaseYarnAPI(object):
     __logger = None
     response_class = Response
 
-    def __init__(self, address=None, port=None, timeout=None, kerberos_enabled=None):
-        self.address, self.port, self.timeout, self.kerberos_enabled = address, port, timeout, kerberos_enabled
+    def __init__(self, service_endpoint=None, timeout=None, auth=None, verify=True):
+        self.timeout = timeout
+
+        if service_endpoint:
+            # Snippet to decouple parts of Uri for further building of URLs
+            self.service_uri = urlparse(service_endpoint)
+            self.service_uri_scheme = self.service_uri.scheme or 'http'
+            self.service_uri_hostname = self.service_uri.hostname or self.service_uri.path
+
+            if self.service_uri.port:
+                self.service_uri_postfix = ':' + str(self.service_uri.port)
+            else:
+                self.service_uri_postfix = ''
+        else:
+            self.service_uri = None
+            self.service_uri_scheme = None
+            self.service_uri_hostname = None
+            self.service_uri_postfix = None
+
+        self.session = requests.Session()
+        self.session.auth = auth
+        self.session.verify = verify
 
     def _validate_configuration(self):
-        if self.address is None:
-            raise ConfigurationError('API address is not set')
-        elif self.port is None:
-            raise ConfigurationError('API port is not set')
+        if not self.service_uri:
+            raise ConfigurationError('API endpoint is not set')
+
+    def _contruct_url(self, api_path):
```

Comment:
Move to `to_url` method per previous comment.

Reply:
Adjusted